hs_core/views/__init__.py

```diff
@@ -697,3 +697,18 @@ def _set_resource_sharing_status(request, user, resource, is_public, is_discover
         # set isPublic metadata AVU accordingly
         istorage = IrodsStorage()
         istorage.setAVU(resource.short_id, "isPublic", str(resource.raccess.public))
+
+
+def _get_message_for_setting_resource_flag(resource, resource_flag):
+    has_files = resource.has_required_content_files()
+    has_metadata = resource.metadata.has_all_required_elements()
+    msg = ''
+    if not has_metadata and not has_files:
+        msg = "Resource may not have sufficient required metadata and/or content files to be {flag}".format(
```

Comment:
in the msg, 'and/or' should be changed to 'and' since the resource does not have required metadata AND content files in this case. Also, should we change "may not" to "does not" to be more assertive to reflect the fact?